linkml/generators/excelgen.py

```diff
@@ -37,49 +37,41 @@ def create_workbook(self, workbook_name: str) -> Workbook:
         :return: An openpyxl Workbook object representing the newly created workbook.
         """
         workbook = Workbook()
-        workbook.title = workbook_name
+        workbook.save(workbook_name)
         return workbook
 
-    def get_workbook_name(self, workbook: Workbook) -> str:
+    def create_workbook_and_worksheets(self, output_path: str, classes: List[str]) -> None:
         """
-        Returns the name of the given workbook.
+        Creates a workbook with worksheets for each class.
 
-        :param workbook: The workbook whose name should be returned.
-        :return: Name of the workbook.
-        """
-        return workbook.title
-
-    def remove_worksheet_by_name(self, workbook: Workbook, worksheet_name: str) -> None:
-        """
-        Remove worksheet from workbook by name.
+        :param output_path: The path where the workbook should be created.
+        :param classes: List of class names for which worksheets should be created.
         """
-        worksheet = workbook[worksheet_name]
-        workbook.remove(worksheet)
-
-    def create_worksheet(self, workbook: Workbook, worksheet_name: str) -> Worksheet:
-        """
-        Creates an Excel worksheet with the given name in the given workbook.
+        workbook = self.create_workbook(output_path)
+        workbook.remove(workbook.active)
+        sv = self.schemaview
 
-        :param workbook: The workbook to which the worksheet should be added.
-        :param worksheet_name: Name of the worksheet to be created.
-        """
-        worksheet = workbook.create_sheet(worksheet_name)
-        workbook_name = self.get_workbook_name(workbook)
-        workbook.save(workbook_name)
+        for cls_name in classes:
+            cls = sv.get_class(class_name=cls_name, imports=self.mergeimports)
+            if not cls.mixin and not cls.abstract:
+                workbook.create_sheet(cls_name)
 
-        return worksheet
+                # Add columns to the worksheet for the current class
+                slots = [s.name for s in sv.class_induced_slots(cls_name, self.mergeimports)]
+                self.add_columns_to_worksheet(workbook, cls_name, slots)
+                workbook.save(output_path)
 
-    def create_schema_worksheets(self, workbook: str) -> None:
-        """
-        Creates worksheets in a given Excel workbook based on the classes in the
-        schema.
+                # Add enum validation for columns with enum types
+                enum_list = [e_name for e_name, _ in sv.all_enums(imports=self.mergeimports).items()]
```

Comment:
You can simplify to `[e_name for e_name in sv.all_enums(imports=self.mergeimports)]` here and once again 3 lines below.

Reply:
agreed